crypto/src/hash.rs

```diff
@@ -80,19 +79,75 @@ impl Hash {
     pub const fn typed<T>(self) -> HashOf<T> {
         HashOf(self, PhantomData)
     }
+
+    /// Check if least significant bit of `[u8; Hash::LENGTH]` is 1
+    fn is_lsb_1(hash: &[u8; Self::LENGTH]) -> bool {
```

Comment:
so hash will always be an odd number? how does this affect the hash distribution?

Reply:
Most hash functions are designed to work with such pathologies, if we used it as a hash table entry (which we wouldn’t). The Merkle proof makes no assumptions about the parity, so it should work with odd (pun intended) hashes. 

For our purposes it’s more important to think about the hash space than hash collisions. All we sacrifice in this case, however is one single bit.

Reply:
If we really wanted to be pedantic, we would right shift the hash before use, but to me it seems superfluous.

Reply:
> Most hash functions are designed to work with such pathologies, if we used it as a hash table entry (which we wouldn’t). The Merkle proof makes no assumptions about the parity, so it should work with odd (pun intended) hashes. 

I agree, but thought I better bring it up

> If we really wanted to be pedantic, we would right shift the hash before use, but to me it seems superfluous. 

I don't see how that would make any meaningful difference

Reply:
I think it would be uniform distributed across in odd numbers.